src/app/services/geocoder-api.ts

```diff
@@ -0,0 +1,201 @@
+
+class RequestBuilder {
+  // required
+  private _baseUrl: string;
+  private _outputFormat: string;
+  private _addressString: string;
+
+  // optional
+  private _minScore: number;
+  private _maxResults: number;
+  private _outputSRS: number;
+
+  private _echo = false;
+  private _autoComplete = false;
+
+  // search within an area (bounding box)
+  private _bbox: string;  // A bounding box (xmin,ymin,xmax,ymax) that limits the search area
+
+  // search within an area (centre point + distance in meters)
+  private _centre: string;  // The coordinates of a centre point (x,y) used to define a bounding circle that will limit the search area
+  private _maxDistance: number;
+
+  constructor() {
+    this._baseUrl = 'https://geocoder.api.gov.bc.ca';
```

Comment:
Not sure if we want these kind of hardcoded configs in a const file or an environment param?

Reply:
Would be nice to have as an environment param, something like the solution EPIC had which would be a bit more work. The current api url's are hardcorded in a switch statement as well. A constants file would be the easiest to push in now

Reply:
Added a shared constants file for the api urls, (geocoder and nrpti)